### PR TITLE
Do not throw Exception if chmod() fails

### PR DESCRIPTION
**Pull request by amcsi**
As per https://github.com/zendframework/zend-log/pull/7#issuecomment-189327632

### CODE REVIEW

File: src/Writer/Stream.php [hunk 1]
```diff
@@ -10,9 +10,11 @@
 namespace Zend\Log\Writer;
 
 use Traversable;
+use Zend\Db\Adapter\Exception\ErrorException;
```
Comment (weierophinney) [inline src/Writer/Stream.php]: This should be removed; it's never referenced by the code. (I can do that on merge.)

File: src/Writer/Stream.php [hunk 2]
```diff
@@ -10,9 +10,11 @@
...
 use Zend\Log\Exception;
 use Zend\Log\Formatter\Simple as SimpleFormatter;
 use Zend\Stdlib\ErrorHandler;
+use Zend\Stdlib\Err;
```
Comment (weierophinney) [inline src/Writer/Stream.php]: This should be removed; doesn't exist! (I'll do that on merge.)

File: test/Writer/StreamWriterTest.php [hunk 3]
```diff
@@ -192,10 +192,47 @@ public function testCanSpecifyFilePermsViaChmodOption()
         $this->assertEquals(0664, $this->root->getChild('foo')->getPermissions());
     }
 
+    public function testFilePermsDoNotThrowErrors()
+    {
+        // Make the chmod() override emit a warning.
+        $GLOBALS['chmod_throw_error'] = true;
+
+        $filter    = new \Zend\Log\Filter\Mock();
+        $formatter = new \Zend\Log\Formatter\Simple();
```
Comment (weierophinney) [inline test/Writer/StreamWriterTest.php]: Import the above (I'll do that on merge)

File: test/Writer/StreamWriterTest.php [hunk 4]
```diff
@@ -192,10 +192,47 @@ public function testCanSpecifyFilePermsViaChmodOption()
...
+ * @return bool
+ * @throws \ErrorException  if $GLOBALS['chmod_throw_error'] is truthy.
+ */
+function chmod($filename, $mode) {
+    if (!empty($GLOBALS['chmod_throw_error'])) {
+        trigger_error('some_error', E_WARNING);
+        return false;
+    }
+    return \chmod($filename, $mode);
+}
```
Comment (weierophinney) [inline test/Writer/StreamWriterTest.php:239]: The above should be in a separate file, required conditionally. (I'll do that on merge.)

**Comment by weierophinney** [comment]
Cherry-picked to master for release with 2.7.2.

**Comment by amcsi** [comment]
Ah didn't notice you self-assigned this. I made the fixes too.

**Pull request closed without merging**